### DISCUSSION TITLE
Alien detection and agression low HAB_Airport

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#100] [Q&A] [answered]
I am just kicking around adding an alien troll in mission 2.  Maybe a kill trap only in HAB_Airport Mission 2.

For some reason while I can create an alien entity and spawn him on a trigger he's not being very aggessive 
outside of lots of noise provocation and no kill traps.   In other mods aliens are often permitted in the baggage area where you 
enter from the airlock into mission11.  Even then the alien(s) needs lots of stimulii to spawn.

I suspect something underlying the HAB_Airport scripts are holding the alien backstage and not having it sweep very wide.
I am not sure what to setup to counter this however. 

I'm aware there are few vents, but from making the alien spawn there seem to be enough to get the job done.
Two or three in the first main floor section where you power on the doors.  I tied a spawn to the logic of power on and it works fine. There's one in the next section where the wire is hanging down. There are 2 or 3 in the terminal area where we see the Torrens flyby.
There are several in the baggage area by the generator and at least two upstairs once we get the crowbar.

 I'll settle for making the alien set killtraps at the chokepoints vs being on the ground which 
might be better than active at this stage since I'll need to despawn it when Axel is brought into play anyway.
Looking to hehe "spice up" mission 2 from being utterly boring w/o skipping it.   I will probably spawn some
help for distractions or not hehe.

I went and added a new ALIENTACTICALPOSITION composite where it doesn't exist on this map using Comms as a template.
And appropriate values to trigger on.  The alien is odd.  It's spawning from the vent by the security kisosk but it almost immediately
upvents unless I make more noise.  Once on the ground it doesn't do as it normally does and rush the source of the disturbance
it calmly walks most of the way unless further incitied.  Something is really messing with it's behaviors.
I set a AlienConfigString to reference a variablestring set to EXTREME.  Should kick something on.  I watched it spawn and still 
immediately upvent.  Like it's leashed to the vents.  Or there's a timer firing.

**Comment by MattFiler** [comment, marked as answer]
You can set aggression and other such attributes on characters through some function entities, perhaps try calling some of those on your spawned alien to see if he's a bit more aggro? It could be that something somewhere is checking the mission number via GLOBAL and setting the alien menace-type values to low as you're on an early level. Not too sure.

Have a look for some of the following:
- `CHR_SetMood`
- `NPC_AlienConfig`
- `NPC_SetSenseSet`
- `NPC_SetupMenaceManager`
- `NPC_SenseLimiter`
- `NPC_SetAlienDevelopmentStage`

**Reply by ghramsey** [reply]
Cool. I'll look through those some more.  I setup the NPC_AlienConfig already.   
Actually created a new instance of it since the other one is linked to the Axel death and post-death portions of the level.

The others I need to check now too.  Some are tied in with the exisiting end of mission 2 cutscene.  
I need to beware to not interfere with or break that which is why I've kinda trod slowly and carefully.

Thanks again.
